src/spec-configuration/containerFeaturesConfiguration.ts

```diff
@@ -433,19 +453,17 @@ function featuresToArray(config: DevContainerConfig): DevContainerFeature[] | un
 
 // Process features contained in devcontainer.json
 // Creates one feature set per feature to aid in support of the previous structure.
-async function processUserFeatures(output: Log, userFeatures: DevContainerFeature[], featuresConfig: FeaturesConfig) : Promise<FeaturesConfig>
-{
-	userFeatures.forEach(userFeature => {
-			const newFeatureSet = parseFeatureIdentifier(output, userFeature);
+async function processUserFeatures(output: Log, env: NodeJS.ProcessEnv, userFeatures: DevContainerFeature[], featuresConfig: FeaturesConfig): Promise<FeaturesConfig> {
+	for (const userFeature of userFeatures) {
+		const newFeatureSet = await parseFeatureIdentifier(output, env, userFeature);
 			if(newFeatureSet) {
 				featuresConfig.featureSets.push(newFeatureSet);
 			}
-		}
-	);
+	}
 	return featuresConfig;
 }
 
-export function parseFeatureIdentifier(output: Log, userFeature: DevContainerFeature) : FeatureSet | undefined {
+export async function parseFeatureIdentifier(output: Log, env: NodeJS.ProcessEnv, userFeature: DevContainerFeature): Promise<FeatureSet | undefined> {
```

Comment:
Should we update the large comment below to reflect the new OCI format? I would also suggest to perhaps move `getFeatureIdType` up from the bottom of the file, to have it closer to this comment since its logic follows this comment.

Reply:
done and moved! Also linking to our spec now